### ISSUE TITLE
Typing issue with langchain, @langchain/core, and @langchain/openai

### ISSUE THREAD

**Issue by AumeshMisra** [langchain-ai/langchainjs#5736]
### Checked other resources

- [X] I added a very descriptive title to this issue.
- [X] I searched the LangChain.js documentation with the integrated search.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I am sure that this is a bug in LangChain.js rather than my code.
- [X] The bug is not resolved by updating to the latest stable version of LangChain (or the specific integration package).

### Example Code

const prompt = PromptTemplate.fromTemplate(myPrompt);
    const model = new AzureOpenAI({
      modelName: "gpt-4o",
      temperature: 0.0,
    });

    const combineDocsChain = await createStuffDocumentsChain({
      llm: model,
      prompt,
    });

    const chain = await createRetrievalChain({
      retriever,
      combineDocsChain,
    });

### Description

I am trying to use createStuffDocumentsChain but it's unable to resolve types. 

For the llm param, I get:
```
Type 'AzureOpenAI' is not assignable to type 'LanguageModelLike'.
  Property 'lc_runnable' is protected but type 'Runnable<RunInput, RunOutput, CallOptions>' is not a class derived from 'Runnable<RunInput, RunOutput, CallOptions>'.
```

for the prompt param, I get:
`The expected type comes from property 'prompt' which is declared here on type '{ llm: LanguageModelLike; prompt: BasePromptTemplate<any, BasePromptValueInterface, any>; outputParser?: BaseOutputParser<...> | undefined; documentPrompt?: BasePromptTemplate<...> | undefined; documentSeparator?: string | undefined; }'`

This was fine when I had:
"@langchain/community": "^0.2.4",
    "@langchain/core": "^0.2.2",
    "@langchain/openai": "^0.0.34",
    
 But now there's an error when I have:
 "@langchain/community": "^0.2.4",
    "@langchain/core": "^0.2.6",
    "@langchain/openai": "^0.1.2",
    
  I even made sure to add:
  "langchain": "^0.2.5",
but I'm still getting issues there.

### System Info

"@langchain/core": "^0.2.6",
    "@langchain/openai": "^0.1.2",
    "@langchain/pinecone": "0.0.6",
    "@pinecone-database/pinecone": "^2.2.2",
    "@prisma/client": "^5.14.0",
    "axios": "^1.6.8",
    "body-parser": "^1.20.2",
    "cors": "^2.8.5",
    "dotenv": "^16.4.5",
    "express": "^4.19.2",
    "langchain": "^0.2.5",

**Comment by jacoblee93** [comment]
Hey @AumeshMisra, sorry you're running into this - can you try setting your resolutions field like this to ensure that only one version of `@langchain/core` is present?

https://js.langchain.com/v0.2/docs/how_to/installation/#installing-integration-packages

CC @bracesproul

**Comment by ik4Rus** [comment]
Hi, 
I run into an issue, that might be related. 
I am following this tutorial [here](https://js.langchain.com/v0.2/docs/how_to/few_shot_examples_chat/#fixed-examples):
```ts
    const examplePrompt = ChatPromptTemplate.fromMessages([
      ['human', '{input}'],
      ['ai', '{output}'],
    ]);

    const fewShotPrompt = new FewShotChatMessagePromptTemplate({
      examplePrompt,
      examples: examplesFewShot,
      inputVariables: [],
    });

    const finalPrompt = ChatPromptTemplate.fromMessages([
      ['system', systemTemplateTaskClassificationFewShot],
      fewShotPrompt,
      ['human', '{input}'],
    ]);
```
My package.json looks like this: 
```
 "dependencies": {
    "langchain": "^0.2.7",
    "@langchain/azure-openai": "^0.0.11",
  },
  ...,
  "resolutions": {
    "@langchain/core": "0.2.09"
  }
```
I get the following TS error for using `FewShotChatMessagePromptTemplate` in `ChatPromptTemplate.fromMessages`: 
```
Type 'FewShotChatMessagePromptTemplate<any, any>' is not assignable to type 'ChatPromptTemplate<InputValues, string> | BaseMessagePromptTemplateLike'.
  Type 'FewShotChatMessagePromptTemplate<any, any>' is missing the following properties from type 'ChatPromptTemplate<InputValues, string>': promptMessages, _parseImagePromptsts(2322)
```
I would be very grateful for any input. I don't see where the usage is different from the tutorial. 

Thank you very much!

**Comment by jacoblee93** [comment]
@ik4Rus can you try without the `0` in front of `9` for resolutions?

So:
```
  "resolutions": {
    "@langchain/core": "0.2.9"
  }
```

**Comment by jacoblee93** [comment]
This should be resolved in 0.3.0 - core is now a peer dep:

https://js.langchain.com/docs/how_to/installation/#installing-integration-packages

If this arises again after updating, please reopen!

**Comment by xjyribro** [comment]
@jacoblee93 Facing this issue after following the tutoial app and [#installing-integration-packages](https://js.langchain.com/docs/how_to/installation/#installing-integration-packages) mentioned above.

```
"dependencies": {
    "@langchain/community": "^0.3.30",
    "@langchain/core": "^0.3.0",
    "@langchain/langgraph": "^0.2.46",
    "@langchain/openai": "^0.4.4",
    "dotenv": "^16.4.7",
    "drizzle-kit": "^0.30.4",
    "drizzle-orm": "^0.39.3",
    "express": "^4.21.2",
    "firebase-admin": "^13.1.0",
    "langchain": "0.0.207",
    "pg": "^8.13.3",
    "serverless-http": "^3.2.0"
  },
  "devDependencies": {
    "@trivago/prettier-plugin-sort-imports": "^5.2.2",
    "@types/express": "^5.0.0",
    "cross-env": "^7.0.3",
    "esbuild": "^0.25.0",
    "prettier": "^3.5.0",
    "tsx": "^4.19.2"
  },
  "pnpm": {
    "overrides": {
      "@langchain/core": "0.3.0"
    }
  }
```

```

  const agentTools = [new TavilySearchResults({ maxResults: 3 })];
  const agentModel = new ChatOpenAI({ temperature: 0 });

  const agentCheckpointer = new MemorySaver();
  const agent = createReactAgent({
    llm: agentModel, <---------------------------
    tools: agentTools,
    checkpointSaver: agentCheckpointer,
  });
```

```
Type 'ChatOpenAI<ChatOpenAICallOptions>' is missing the following properties from type 'Runnable<BaseLanguageModelInput, LanguageModelOutput, RunnableConfig>': lc_runnable, getName, invoke, bind, and 25 more.ts(2740)
```
